pyperf/tests/test_runner.py

```diff
@@ -149,6 +152,33 @@ def test_pipe(self):
         self.assertEqual(bench_json,
                          tests.benchmark_as_json(result.bench))
 
+    def test_pipe_with_timeout(self):
+        rpipe, wpipe = create_pipe()
+        with rpipe:
+            with wpipe:
+                arg = wpipe.to_subprocess()
+                # Don't close the file descriptor, it is closed by
+                # the Runner class
+                wpipe._fd = None
+
+                result = self.exec_runner('--pipe', str(arg),
+                                          '--worker', '-l1', '-w1')
+
+            # Mock the select to make the read pipeline not ready
+            with mock.patch('pyperf._utils.select.select',
+                            return_value=(False, False, False)):
+                with self.assertRaises(TimeoutError) as cm:
+                    rpipe.read_text(timeout=0.1)
+                self.assertEqual(str(cm.exception),
+                         'Timed out after 0.1 seconds')
+
+            # Mock the select to make the read pipeline ready
+            with mock.patch('pyperf._utils.select.select',
+                            return_value=(True, False, False)):
+                bench_json = rpipe.read_text(timeout=0.1)
+                self.assertEqual(bench_json,
+                                tests.benchmark_as_json(result.bench))
```

Comment:
The test fails on Windows because of different newline:

```
'(...)"version":"1.0"}\r\n' != '(...)"version":"1.0"}\n'
```

You can strip newlines in the comparison:

```suggestion
                self.assertEqual(bench_json.rstrip(),
                                 tests.benchmark_as_json(result.bench).rstrip())
```

Reply:
Done.